src/test/java/org/alfresco/repo/event2/EventConsolidatorUnitTest.java

```diff
@@ -0,0 +1,303 @@
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.event2;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import junit.framework.TestCase;
+import org.alfresco.model.ContentModel;
+import org.junit.Test;
+
+public class EventConsolidatorUnitTest extends TestCase
+{
+    private NodeResourceHelper nodeResourceHelper = mock(NodeResourceHelper.class);
+    
+    @Test
+    public void testGetMappedAspectsBeforeRemovedAndAddedEmpty()
+    {
+        EventConsolidator eventConsolidator = new EventConsolidator(nodeResourceHelper);
+        
+        Set<String> currentApects = new HashSet<>();
+        currentApects.add("cm:geographic");
+        currentApects.add("cm:auditable");
+
+        Set<String> mappedAspectsBefore = eventConsolidator.getMappedAspectsBefore(currentApects);
+
+        assertEquals(0, mappedAspectsBefore.size());
+    }
+
+    @Test
+    public void testGetMappedAspectsBefore_AspectRemoved()
+    {
+        EventConsolidator eventConsolidator = new EventConsolidator(nodeResourceHelper);
+        eventConsolidator.addAspect(ContentModel.ASSOC_CONTAINS);
+
+        Set<String> currentApects = new HashSet<>();
+        currentApects.add("cm:geographic");
+        currentApects.add("cm:auditable");
+
+        Set<String> removed = new HashSet<>();
+        Set<String> added = new HashSet<>();
+        removed.add("cm:contains");
+
+        when(nodeResourceHelper.mapToNodeAspects(eventConsolidator.getAspectsRemoved())).thenReturn(removed);
+        when(nodeResourceHelper.mapToNodeAspects(eventConsolidator.getAspectsAdded())).thenReturn(added);
+
+        Set<String> mappedAspectsBefore = eventConsolidator.getMappedAspectsBefore(currentApects);
+
+        assertEquals(3, mappedAspectsBefore.size());
+        System.out.println(mappedAspectsBefore);
+    }
+
+    @Test
+    public void testGetMappedAspectsBefore_AspectAdded()
+    {
+        EventConsolidator eventConsolidator = new EventConsolidator(nodeResourceHelper);
+        eventConsolidator.addAspect(ContentModel.ASSOC_CONTAINS);
+
+        Set<String> currentApects = new HashSet<>();
```

Comment:
Same typo. I'm not going to add comment for others, so please make sure all `currentApects` variables are changed to `currentAspects`

Reply:
Done